devito/propagator.py

```diff
@@ -35,10 +37,11 @@ class Propagator(object):
                       a list stating block sizes for each dimension. Set block size to None to skip blocking on that dim
     """
 
-    def __init__(self, name, nt, shape, spc_border=0, time_order=0,
+    def __init__(self, name, nt, shape, stencils, factorized=[], spc_border=0, time_order=0,
```

Comment:
Never use ``[]`` for initialisation. Use ``None`` and then in the constructor you can do

```
self.factorized = factorized or []
```

Explanation here: http://docs.python-guide.org/en/latest/writing/gotchas/